src/main/java/com/configcat/ConfigCatClient.java

```diff
@@ -129,6 +129,61 @@ public <T> CompletableFuture<T> getValueAsync(Class<T> classOfT, String key, Use
                 .thenApply(settingResult -> this.getValueFromSettingsMap(classOfT, settingResult, key, user, defaultValue));
     }
 
+    @Override
+    public <T> EvaluationDetails<T> getValueDetails(Class<T> classOfT, String key, T defaultValue) {
+        return this.getValueDetails(classOfT, key, null, defaultValue);
+    }
+
+    @Override
+    public <T> EvaluationDetails<T> getValueDetails(Class<T> classOfT, String key, User user, T defaultValue) {
+        if (key == null || key.isEmpty())
```

Comment:
`user` argument should be checked for null and the `defaultUser` should be taken into account here.

Reply:
It is handled in the `getValueDetailsAsync` method, or do you mean in the catch as well? I would leave the original `user` argument in the catchs.

Reply:
Ah, ok, sry, I missed that part.